mods/default/tools.lua

```diff
@@ -24,383 +24,242 @@ minetest.register_item(":", {
+		punch = 1.3, drop = 0, damage = 3,
+		cap = {times={[2]=2.0, [3]=1.00}, uses=20, maxlevel=1},
+		groups = {pickaxe = 1},
 	},
-	sound = {breaks = "default_tool_breaks"},
-	groups = {pickaxe = 1}
-})
-
-minetest.register_tool("default:pick_bronze", {
-	description = S("Bronze Pickaxe"),
-	inventory_image = "default_tool_bronzepick.png",
-	tool_capabilities = {
-		full_punch_interval = 1.0,
-		max_drop_level=1,
-		groupcaps={
-			cracky = {times={[1]=4.50, [2]=1.80, [3]=0.90}, uses=20, maxlevel=2},
-		},
-		damage_groups = {fleshy=4},
+	bronze = {
+		desc = "Bronze Pickaxe",
+		punch = 1.0, drop = 1, damage = 4,
+		cap = {times={[1]=4.50, [2]=1.80, [3]=0.90}, uses=20, maxlevel=2},
+		groups = {pickaxe = 1},
 	},
-	sound = {breaks = "default_tool_breaks"},
-	groups = {pickaxe = 1}
-})
-
-minetest.register_tool("default:pick_steel", {
-	description = S("Steel Pickaxe"),
-	inventory_image = "default_tool_steelpick.png",
-	tool_capabilities = {
-		full_punch_interval = 1.0,
-		max_drop_level=1,
-		groupcaps={
-			cracky = {times={[1]=4.00, [2]=1.60, [3]=0.80}, uses=20, maxlevel=2},
-		},
-		damage_groups = {fleshy=4},
+	steel = {
+		desc = "Steel Pickaxe",
+		punch = 1.0, drop = 1, damage = 4,
+		cap = {times={[1]=4.00, [2]=1.60, [3]=0.80}, uses=20, maxlevel=2},
+		groups = {pickaxe = 1},
 	},
-	sound = {breaks = "default_tool_breaks"},
-	groups = {pickaxe = 1}
-})
-
-minetest.register_tool("default:pick_mese", {
-	description = S("Mese Pickaxe"),
-	inventory_image = "default_tool_mesepick.png",
-	tool_capabilities = {
-		full_punch_interval = 0.9,
-		max_drop_level=3,
-		groupcaps={
-			cracky = {times={[1]=2.4, [2]=1.2, [3]=0.60}, uses=20, maxlevel=3},
-		},
-		damage_groups = {fleshy=5},
+	mese = {
+		desc = "Mese Pickaxe",
+		punch = 0.9, drop = 3, damage = 5,
+		cap = {times={[1]=2.4, [2]=1.2, [3]=0.60}, uses=20, maxlevel=3},
+		groups = {pickaxe = 1},
 	},
-	sound = {breaks = "default_tool_breaks"},
-	groups = {pickaxe = 1}
-})
+	diamond = {
+		desc = "Diamond Pickaxe",
+		punch = 0.9, drop = 3, damage = 5,
+		cap = {times={[1]=2.0, [2]=1.0, [3]=0.50}, uses=30, maxlevel=3},
+		groups = {pickaxe = 1},
+	}
+}
 
-minetest.register_tool("default:pick_diamond", {
-	description = S("Diamond Pickaxe"),
-	inventory_image = "default_tool_diamondpick.png",
-	tool_capabilities = {
-		full_punch_interval = 0.9,
-		max_drop_level=3,
-		groupcaps={
-			cracky = {times={[1]=2.0, [2]=1.0, [3]=0.50}, uses=30, maxlevel=3},
+for mat, tool in pairs(tool_pick) do
+	minetest.register_tool("default:pick_".. mat, {
+		description = S(tool.desc),
+		inventory_image = "default_tool_".. mat .."pick.png",
+		tool_capabilities = {
+			full_punch_interval = tool.punch,
+			max_drop_level = tool.drop,
+			groupcaps={
+				cracky = tool.cap,
+			},
+			damage_groups = {fleshy = tool.damage},
 		},
-		damage_groups = {fleshy=5},
-	},
-	sound = {breaks = "default_tool_breaks"},
-	groups = {pickaxe = 1}
-})
+		sound = {breaks = "default_tool_breaks"},
+		groups = tool.groups
```

Comment:
You can add
`tool.groups.pickaxe = 1`
before `minetest.register_tool`

Reply:
Alternatively maybe I could do this?
```
        tool.groups = {pickaxe = 1}
        if mat == "wood" then
                tool.groups.flammable = 2
        end
```
Then remove `groups` from the trables, but this might be taking it too far?

Reply:
Or even.
```
local function tool_wood_groups(mat, amount)
        local groups = {}
        if mat == "wood" then
                groups.flammable = amount
        end
        return groups
end

tool.groups = tool_wood_groups(mat, 2)
tool.groups.pickaxe = 1
```

Reply:
Do you want to simplify, or make it more complicated?)

Reply:
Point taken, I changed it to how I think you meant.

Reply:
You can remove
groups = {} and add
tool.groups = tool.groups or {}

Reply:
Thanks for pointing out what should of been obvious. :)